### ISSUE TITLE
Hotkey to navigate to a spesified cell in a table

### ISSUE THREAD

**Issue by dangmanhcuong87** [nvaccess/nvda#11232]
### Is your feature request related to a problem? Please describe.
When work with a table contains multiple rows and columns, It would be slow if I want to read data from a cell such as row 50, column 30. I must hold Ctrl+Alt, and use arrow keys to jump to that cell.
### Describe the solution you'd like
Provide a dialog so people could call it by a hotkey, and enter cell coordinates to quickly navigate to a cell.
### Describe alternatives you've considered
Save position of a table cell for a file, or a page contains the table, like bookmark.

**Comment by XLTechie** [comment]
I also work with a lot of really huge tables, and could see substantial value in 
this.

**Comment by Adriani90** [comment]
If you have huge tables, you also know actually rather the row title that you want to move to, not the cell coordinate. So actually you can search for that with ctrl+nvda+f and jump directly to the row you want, or you can search for the value of the cell if you know it. In MS excel you can jump to a specified cell address. To do this, press ctrl+g and type the address you want (i.e. g50).

Other than MS Excel or Libre Calc, it is not common at all to work with cell coordinates. So I don't see any clear justification for this feature given the possibilities that we already have.

**Comment by XLTechie** [comment]
For me, I was talking about HTML tables.

And in the cases where I usually run into them (system administration, program 
code presented in tabular form for some stupid reason, virtual machine instances 
and their associated services), there is no way to know what a cell might 
contain before getting to it.

**Comment by CyrilleB79** [comment]
@Adriani90 the assumption that the user knows the row title is not always correct IMO.
In the case of tables containing only numeric data with the date/time in the first cell of the row, the user does not always know the exact date/time of a line.

A common use case of this feature would be when the user explores a huge table of numeric data. And if he want to see what the data look like in the middle of the table. He will ask to jump to a cell whose coordinates are located about at half of the table size.

I do not know if the solution proposed in this issue is the best one. But for sure, NVDA lacks commodity to explore huge tables of numeric data.

**Comment by Adriani90** [comment]
Rather than implementing a new dialog for searching for cell coordinate, I would suggest to implement an option which works only in browse mode and which lets you define how many rows and columns are skipped when pressing ctrl+alt+arrow keys. I.e. an edit field for rows and an edit field for columns in which you can enter a numeric value (i.e. 10 rows and 2 columns). That means, NVDA would skip 10 rows when pressing ctrl+alt+down arrow or two columns when pressing ctrl+alt+left arrow. And you would only need a gesture which would restore the default setting which is navigation row by row and column by column.
This should make table navigation easier and exploring a huge table would be much more efficient.
A gesture for restore the default behavior could be ctrl+alt+1 for example.

**Comment by Adriani90** [comment]
I am struggling with navigating to a certain cell coordinate because for a sighted person this information is not available actually, so people scroll the table if they explore it. So we should adopt the scrolling principle for a screen reader as well.
The setting suggestion I mentioned above should be profile specific and could be located in browse mode settings.

**Comment by Adriani90** [comment]
Only in Excel or Libre Calc, cell coordinates are also used by sighted people to navigate through the tables.

**Comment by CyrilleB79** [comment]
> I am struggling with navigating to a certain cell coordinate because for a sighted person this information is not available actually, so people scroll the table if they explore it. So we should adopt the scrolling principle for a screen reader as well.

I agree that a screen reader does not aim to provide natively general features that would also benefit to sighted people. But trying to compensate by any mean the big advantage a sighted person has over a blind one in table exploration would clearly justify the suggestion of @dangmanhcuong87 .

Actually, scrolling during table exploration covers 2 use cases:
* Read partially but more quickly the content of a table: this use case is covered by @Adriani90's suggestion. This would be a sort of skim reading for tables.
* Jumping as quickly as possible to a location much farther in the table: this use case would be covered by @dangmanhcuong87's suggestions.

**Comment by XLTechie** [comment]
I'm not sure I understand why this method is better than the original proposed 
direct jump method.

Instead of directly solving the problem, this proposed method:

* Does introduce a new dialog, in order to choose skip interval.
* Does have to add two gestures: one to invoke dialog, and one to reset.
* Requires the user to do mental math or Monte Carlo method jumps and reversals to find the desired area.

(And, from your later message about profiles, I am guessing)

* Requires the user, instead of quickly just entering a scroll position, to go to 
some place in NVDA settings, make the change (hopefully in profile and not 
globally), and remember to have the right profile loaded.

Does doing it this way solve some other use case as well, that the first 
suggestion misses?

**Comment by JulienCochuyt** [comment]
> IIn MS excel you can jump to a specified cell address. To do this, press ctrl+g and type the address you want (i.e. g50).

Mimicking this feature for browseable documents (eg. HTML, MS Word, PDF, ...) tables sounds pretty reasonable to me.

@dangmanhcuong87 wrote:
> Save position of a table cell for a file, or a page contains the table, like bookmark.

This can be achieved with WebAccess add-on, and maybe also with PlaceMarker add-on. (I'm not sure about the latter, though.)

Still, I feel your first described solution is valuable for the general case.
I'll try to find the time to propose an implementation as soon as the #11006 hold is over.

**Comment by dangmanhcuong87** [comment]
In Excel it very Easy, but in other program I still keep thinking of this suggestion because most of them don't have cell address.

**Comment by bhavyashah** [comment]
A few questions below:
* To emulate the ability of sighted people scrolling through a table, why is pressing and holding down Ctrl+Alt+arrow keys an insufficient equivalent? In fact, if a three finger seems like a hassle, you can use the Easy Table Navigator add-on to toggle to a mode wherein you can move rapidly through a table using only the arrow keys.
* How common is it for users to deal with large tables in programs other than Excel or Libre Calc, i.e. programs wherein you do not have an existing Go To Cell functionality? Please list several use cases and real-world scenarios.
* Currently, NVDA offers a Find dialog but not a Go TO Cell dialog. The requested change is based on the logic that there are instances wherein a user does not know the contents of the cell they are looking for but knows its coordinates. This doesn't strike me as being a common occurrence, because it seems to me that most people would know the data or text they are searching for and could just use the Find functionality to jump to the cell containing it. Could you please explain what I am missing here?
I think answering these questions one by one might help bring everyone on the same page. @dangmanhcuong87 @JulienCochuyt @XLTechie @CyrilleB79 @Adriani90

**Comment by Adriani90** [comment]
@bhavyashah big tables are also common in browsers, especially when you are working with data bases like in SQL. Regarding your other questions, I think the request is search for a specific cell address, to make scrolling easier and to scroll the table radomly, i.e. you want to see what's in the middle on row 300 of 600, or you want to see what in the row 150, column 6. This might be helpful for numeric data bases where the cell content is really hard to remember.

**Comment by Adriani90** [comment]
@bhavyashah one more hint from my side, if a feature is provided by an addon this does not mean that an issue on Github is solved. There are lot of people out there with no or weak internet connection in developing countries who cannot download addons, there are also many corporate environment which don't allow to download and install addons etc. It is worthy to mention that an issue is solved by an addon for some people, but it is not a reason to close an issue in general in my opinion.

**Comment by bhavyashah** [comment]
@Adriani90  Fully agreed. I was just thinking if the add-on's functionality adequately addressed the problem, we could integrate that solution into core itself rather than the range of other solutions proposed throughout this thread.
